### PR TITLE
probert: use os-prober only if asked

### PR DESCRIPTION
**Pull request by dbungert**
* add command line arg for using os-prober
* use probe_types to pass message along

### CODE REVIEW

**Comment by dbungert** [comment]
@jpnurmi - when this merges, ubuntu-desktop-installer will want https://github.com/canonical/ubuntu-desktop-installer/pull/749

File: subiquity/cmd/server.py [hunk 1]
```diff
@@ -73,6 +73,8 @@ def make_server_args_parser():
         help='in dryrun, control basedir of files')
     parser.add_argument(
         '--storage-version', action='store', type=int, default=1)
+    parser.add_argument(
+        '--use-os-prober', action='store', type=bool, default=False)
```
Comment (ogayot) [inline subiquity/cmd/server.py:77]: I don't think this does what you want, or does it?
`action='store'` indicates that the next argument must be consumed and converted into a boolean, e.g.:

```python
server.py --use-os-prober foo     # bool("foo") => True
server.py --use-os-prober ""      # bool("") => False
server.py --use-os-prober False   # bool("False") => True
server.py --use-os-prober True    # bool("True") => True
server.py --use-os-prober         # -> will fail / missing argument
```

I think this is what you want instead:
```python
parser.add_argument(
    '--use-os-prober', action='store_true')
```

**Review by ogayot — CHANGES_REQUESTED** [review]

File: subiquity/server/controllers/tests/test_filesystem.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Copyright 2019 Canonical, Ltd.
```
Comment (ogayot) [inline subiquity/server/controllers/tests/test_filesystem.py:1]: 2022 :D

File: subiquity/cmd/server.py (hunk 1 above)
Comment (dbungert) [reply subiquity/cmd/server.py:77]: Yes, I wanted store_true.

**Review by ogayot — APPROVED** [review]

File: snapcraft.yaml [hunk 3]
```diff
@@ -183,9 +183,9 @@ parts:
       - libnl-3-dev
       - libnl-genl-3-dev
       - libnl-route-3-dev
-    source: https://github.com/canonical/probert.git
+    source: https://github.com/dbungert/probert.git
```
Comment (mwhudson) [inline snapcraft.yaml:186]: Not this bit :)

**Review by mwhudson — APPROVED** [review]

**Pull request merged**